modin/pandas/groupby.py

```diff
@@ -878,14 +1060,11 @@ def _wrap_aggregation(
         """
         if self._axis != 0:
             return self._default_to_pandas(default_func, **kwargs)
-        # For aggregations, pandas behavior does this for the result.
-        # For other operations it does not, so we wait until there is an aggregation to
-        # actually perform this operation.
-        if not self._is_multi_by and drop and self._drop and self._as_index:
-            groupby_qc = self._query_compiler.drop(columns=self._by.columns)
```

Comment:
dropping 'by' columns is redundant, we will insert them back anyway at the backend in order to ensure the existence of the 'by' columns in every partition, removing these lines saves one extra concatenation